pkg/virt-controller/watch/workload-updater/workload-updater_test.go

```diff
@@ -186,8 +186,7 @@ var _ = Describe("Workload Updater", func() {
 				reasons = append(reasons, SuccessfulEvictVirtualMachineInstanceReason)
 			}
 
-			// wait for informer to catch up since we aren't watching for vmis directly
-			time.Sleep(1 * time.Second)
+			waitForNumberOfInstancesOnVMIInformerCache(controller, 200)
```

Comment:
I think it would increase the overall test's readability if you had a counter where you "put" the number of VMs you're creating, and finally wait for that particular number of VMs.

Something like:
```golang

totalVMs := 0
for i := 0; i < 50; i++ {
    newVirtualMachine(fmt.Sprintf("testvm-migratable-%d", i), true, "madeup", vmiSource, podSource)
    totalVMs++
}

for i := 0; i < 50; i++ {
    newVirtualMachine(fmt.Sprintf("testvm-non-migratable-%d", i), false, "madeup", vmiSource, podSource)
    totalVMs++
}

// add vmis that are not outdated to ensure they are not counted as outdated in count
for i := 0; i < 100; i++ {
    newVirtualMachine(fmt.Sprintf("testvm-up-to-date-%d", i), false, expectedImage, vmiSource, podSource)
    totalVMs++
}
...

waitForNumberOfInstancesOnVMIInformerCache(controller, totalVMs)
```

Reply:
Thanks! great suggestion!